### PR TITLE
JBIDE-12532: Fixed Open URL in internal browser

### PR DESCRIPTION
**Pull request by gastaldi**

### CODE REVIEW

**Comment by gastaldi** [comment]
@fbricon @maxandersen can any of you guys review this PR? Thanks

**Comment by lincolnthree** [comment]
It looks like this only works for URLs. Why not support filesystem resources as well?

**Comment by gastaldi** [comment]
That's what the browse() method is for. For filesystem resources, there is the edit() and the open() methods

**Comment by maxandersen** [comment]
is this a regression in 4.2.2 or also present in 4.2.0 and 4.2.1 ?

File: plugins/org.jboss.tools.forge.ui/src/org/jboss/tools/forge/ui/internal/ext/provider/ForgeUIDesktop.java [hunk 1]
```diff
@@ -70,17 +71,23 @@ public void run() {
...
-					.createBrowser(ForgeUIPlugin.PLUGIN_ID);
-			browser.openURL(uri.toURL());
-		} catch (PartInitException e) {
-			Status status = new Status(IStatus.ERROR, ForgeUIPlugin.PLUGIN_ID,
-					"Browser initialization failed");
-			ForgeUIPlugin.log(e);
-			MessageDialog.openError(null, "Browse URL", status.getMessage());
-		}
+		final URL url = uri.toURL();
+		Display.getDefault().syncExec(new Runnable() {
```
Comment (maxandersen) [inline plugins/org.jboss.tools.forge.ui/src/org/jboss/tools/forge/ui/internal/ext/provider/ForgeUIDesktop.java]: what is it fixing that it is run in sync ?

**Comment by gastaldi** [comment]
Also present in 4.2.1, but a different exception

File: plugins/org.jboss.tools.forge.ui/src/org/jboss/tools/forge/ui/internal/ext/provider/ForgeUIDesktop.java (hunk 1 above)
Comment (gastaldi) [inline plugins/org.jboss.tools.forge.ui/src/org/jboss/tools/forge/ui/internal/ext/provider/ForgeUIDesktop.java]: The DefaultBrowserSupport class from Eclipse relies on the active workspace.

**Comment by fbricon** [comment]
+1 to apply

**Pull request merged**